zerver/lib/validator.py

```diff
@@ -52,6 +52,14 @@ def check_capped_string(var_name: str, val: object, max_length: int) -> Optional
 def check_long_string(var_name: str, val: object) -> Optional[str]:
     return check_capped_string(var_name, val, 500)
 
+def check_date(var_name: str, val: object) -> Optional[str]:
+    try:
+        date_arg = [int(i) for i in val.split("-")]
+        date(*date_arg)
+    except ValueError:
+        return _('%s is not a date') % (var_name,)
```

Comment:
I don't like calling a function with user-specified arguments as a general rule; it is a code smell for security bugs.

I think this would be a much safer algorithm for validating the date:
https://stackoverflow.com/questions/16870663/how-do-i-validate-a-date-string-format-in-python